src/state/slices/time_slice.ts

```diff
@@ -48,17 +57,24 @@ export const createTimeSlice: StateCreator<TimeSlice & DatasetSlice, [], [], Tim
     () => get().currentFrame,
     async (frame) => {
       set({ pendingFrame: frame });
-      await get().loadFrameCallback(frame);
-      set({ currentFrame: frame });
+      const result = await get().frameLoadCallback(frame);
+      if (result !== null) {
+        set({ currentFrame: result.frame, frameLoadResult: result });
+      } else if (get().pendingFrame === frame) {
```

Comment:
Will there be cases that `result` is null but `get().pendingFrame !== frame`?

Reply:
Yes. In that case, I expect that another `setFrame` call has happened elsewhere in the app after this load was initialized, so the `pendingFrame` has been set to whatever the other value is. If that's the case, we don't want to touch `pendingFrame` in this handler.

(I think this may be unlikely to happen though because `null` results are returned synchronously... I likely need to make a more robust frame request system in the future, but we haven't encountered any UI issues with this SO far.)